sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/ServerCallAsync.java

```diff
@@ -0,0 +1,511 @@
+        return pauseRecordingWithResponse(recordingId, Context.NONE);
+    }
+
+    Mono<Response<Void>> pauseRecordingWithResponse(String recordingId, Context context) {
+        try {
+            Objects.requireNonNull(recordingId, "'recordingId' cannot be null.");
+            return withContext(contextValue -> {
+                contextValue = context == null ? contextValue : context;
+                return this.serverCallInternal
+                    .pauseRecordingWithResponseAsync(serverCallId, recordingId, contextValue)
+                    .onErrorMap(CommunicationErrorException.class, CallingServerErrorConverter::translateException);
+            });
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Resume recording
+     *
+     * @param recordingId The recording id to stop.
+     * @return response for a successful resumeRecording request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> resumeRecording(String recordingId) {
+        try {
+            Objects.requireNonNull(recordingId, "'recordingId' cannot be null.");
+            return this.serverCallInternal.resumeRecordingAsync(serverCallId, recordingId)
+                .onErrorMap(CommunicationErrorException.class, CallingServerErrorConverter::translateException);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Resume recording
+     *
+     * @param recordingId The recording id to stop.
+     * @return response for a successful resumeRecording request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<Void>> resumeRecordingWithResponse(String recordingId) {
+        return resumeRecordingWithResponse(recordingId, Context.NONE);
+    }
+
+    Mono<Response<Void>> resumeRecordingWithResponse(String recordingId, Context context) {
+        try {
+            Objects.requireNonNull(recordingId, "'recordingId' cannot be null.");
+            return withContext(contextValue -> {
+                contextValue = context == null ? contextValue : context;
+                return this.serverCallInternal
+                    .resumeRecordingWithResponseAsync(serverCallId, recordingId, contextValue)
+                    .onErrorMap(CommunicationErrorException.class, CallingServerErrorConverter::translateException);
+            });
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Get recording state
+     *
+     * @param recordingId The recording id to stop.
+     * @return response for a successful getRecordingState request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<CallRecordingStateResult> getRecordingState(String recordingId) {
+        try {
+            Objects.requireNonNull(recordingId, "'recordingId' cannot be null.");
```

Comment:
Is this a client side or server side param?

Reply:
This is passed from client side, so checking null